File: services/meal-service/src/main/java/com/example/mealservice/api/dto/MealDTO.java
package com.example.mealservice.api.dto;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;


@Builder
public record MealDTO(
        String mealId,
        String name,
        String category,
        String description,
        BigDecimal price,
        List<IngredientForMealDTO> ingredientsForMeal,
        String restaurantId
) {


}

File: FoodOrderingApp/src/models/meal.dto.ts
import { Category } from './category';
import {IngredientForMealDTO} from './meal.ingredient.dto';

export interface MealDTO {
  mealId: string;
  name: string;
  category: Category;
  description: string;
  price: number;
  ingredientsForMeal: IngredientForMealDTO[];
  restaurantId: string;
}

